ratelimit.go

```diff
@@ -171,6 +171,19 @@ func (tb *Bucket) takeAvailable(now time.Time, count int64) int64 {
 	return count
 }
 
+// Available returns the number of available tokens.
+// It will be negative when there are consumers waiting for tokens.
+func (tb *Bucket) Available() int64 {
+	tb.mu.Lock()
+	defer tb.mu.Unlock()
+	return tb.avail
```

Comment:
This is not right. The number of available tokens changes as time progresses
but the returned value from this function does not.

You need to adjust avail before returning the value. Something like
this:
	
	// Available returns the number of available tokens.
	// It will be negative when there are consumers waiting for tokens.
	// Note that this method is inherently racy - the returned
	// count may not be current a moment later.
	func (tb *Bucket) Available(now time.Time) int64 {
		return tb.available(time.Now())
	}
	
	// available is the internal version of available - it takes the current time as
	// an argument to enable easy testing.
	func (tb *Bucket) available(now time.Time) int64 {
		tb.mu.Lock()
		defer tb.mu.Unlock()
		tb.adjust(now)
		return tb.avail
	}

Also, please write a test (note the internal version above, as with the
other methods, written that way so it's easy to write a deterministic test).